components/org.wso2.carbon.identity.application.authenticator.live/src/main/java/org/wso2/carbon/identity/application/authenticator/live/WindowsLiveOAuth2AuthenticatorConstants.java

```diff
@@ -33,8 +35,19 @@
     public static final String GIVEN_NAME_CLAIM_URI = "http://wso2.org/claims/givenname";
     public static final String LAST_NAME_CLAIM_URI = "http://wso2.org/claims/lastname";
     public static final String GENDER_CLAIM_URI = "http://wso2.org/claims/gender";
-    public static final String LOCALITY_CLAIM_URI = "http://wso2.org/claims/locality";
+    public static final String LOCALITY_CLAIM_URI = getClaimUriLocale();
 
     private WindowsLiveOAuth2AuthenticatorConstants() {
     }
+
+    private static String getClaimUriLocale() {
+
+        if (Boolean.parseBoolean(IdentityUtil.getProperty("UseLegacyLocalizationClaim"))) {
+            return "http://wso2.org/claims/locality";
+        } else if (IdentityUtil.getProperty("UseLegacyLocalizationClaim") == null) {
+            return "http://wso2.org/claims/local";
```

Comment:
`Boolean.parseBoolean(IdentityUtil.getProperty("UseLegacyLocalizationClaim"))` returns false when `IdentityUtil.getProperty("UseLegacyLocalizationClaim") == null`. Isn't this redundant?

Reply:
Correct, its redundant
Fixed in - https://github.com/wso2/carbon-identity-framework/pull/3634/commits/1da6face9ef31e4f037c27aff176ef51dd25d85b